src/request-factory.ts

```diff
@@ -1,27 +1,17 @@
 import * as queryString from 'query-string';
+import { Headers } from './header';
+import { RequestOptions } from './request-options';
 
 export default class RequestFactory {
-    /**
-     * @param {string} url
-     * @param {Object} [options={}]
-     * @returns {XMLHttpRequest}
-     */
-    createRequest(url, options = {}) {
+    createRequest(url: string, options: RequestOptions = {}): XMLHttpRequest {
```

Comment:
Can you mark `options` as optional?

Reply:
I tried marking it as optional but it does not allow me to do so if a default is specified